src/stringobj.h

```diff
@@ -281,10 +323,28 @@ extern Int IsStringConv (
 void MakeImmutableString(Obj str);
 
 
-Obj MakeString(const Char *cstr);
+// Functions to create mutable and immutable GAP strings from C strings.
+// MakeString and MakeImmString are inlineable so 'strlen' can be optimised
+// away for constant strings.
```

Comment:
By the way, there is one advantages the `C_NEW_STRING` have over `MakeString` and friends: they are harder to use incorrectly... this code is wrong:
```
ADDR_OBJ(obj)[1] = MakeString("foobar");
```
but `C_NEW_STRING` forces you to create the string before assigning it...